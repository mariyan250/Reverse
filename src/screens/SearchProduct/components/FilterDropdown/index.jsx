// React and Style
import React from 'react';
import './index.scss';

// Translation
import { useTranslation } from 'react-i18next';

// Redux
import { connect } from 'react-redux';
import { filterPost } from 'store/actions/post';

function FilterDropdown({ posts, filterPost }) {
  const { t } = useTranslation();

  const links = [
    {
      text: 'search_screen.filter_dropdown.newest',
      onClick: () => filterPost(posts, 'newest'),
    },
    {
      text: 'search_screen.filter_dropdown.lowest',
      onClick: () => filterPost(posts, 'lowest'),
    },
    {
      text: 'search_screen.filter_dropdown.highest',
      onClick: () => filterPost(posts, 'highest'),
    },
  ];

  return (
    <ul className="filter-dropdown position-absolute app-bg-white rounded-lg box-shadow overflow-hidden d-flex flex-column justify-content-center">
      {links.map((link, i) => (
        <li className="py-3 text-center" key={i} onClick={() => link.onClick()}>
          {t(link.text)}
        </li>
      ))}
    </ul>
  );
}

const mapStateToProps = (state) => ({
  posts: state.posts.items,
});

export default connect(mapStateToProps, { filterPost })(FilterDropdown);
