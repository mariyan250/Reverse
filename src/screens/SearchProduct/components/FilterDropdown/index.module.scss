@import '../../../../index.scss';

.dropdown {
  z-index: 1;
  left: to-rem(-60px);
  top: to-rem(45px);
  width: to-rem(200px);
  animation: appear 1 $timing-fast;
  box-shadow: $shadow-small;

  li {
    font-size: to-rem(16px);

    &:hover {
      background: $silver;
    }
  }
}
