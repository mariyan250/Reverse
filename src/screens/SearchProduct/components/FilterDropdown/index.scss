@import '../../../../index.scss';

.filter-dropdown {
  z-index: 1;
  left: to-rem(-50px);
  top: to-rem(45px);
  width: to-rem(200px);
  animation: appear 1 $fast;

  li {
    font-size: to-rem(16px);

    &:hover {
      background: $silver;
    }
  }
}
