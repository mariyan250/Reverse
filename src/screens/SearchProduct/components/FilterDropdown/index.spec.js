import React from 'react';
import { render, waitForElement } from 'tests/test-utils';
import FilterDropdown from './index';

describe('FilterDropdown', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', async () => {
      const { container, getByTestId } = render(<FilterDropdown />, initialState);
      await waitForElement(() => getByTestId('filter-dropdown'));
      expect(container).toMatchSnapshot();
    });
  });
});
