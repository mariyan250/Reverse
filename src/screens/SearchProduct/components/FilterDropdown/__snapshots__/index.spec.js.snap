// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDropdown Rendering should render the component with initial state 1`] = `
<ul
  className="filter-dropdown position-absolute app-bg-white rounded-lg box-shadow overflow-hidden d-flex flex-column justify-content-center"
  data-testid="filter-dropdown"
>
  <li
    className="py-3 text-center"
    onClick={[Function]}
  >
    search_screen.filter_dropdown.newest
  </li>
  <li
    className="py-3 text-center"
    onClick={[Function]}
  >
    search_screen.filter_dropdown.lowest
  </li>
  <li
    className="py-3 text-center"
    onClick={[Function]}
  >
    search_screen.filter_dropdown.highest
  </li>
</ul>
`;
